analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.CSharp8.cs

```diff
@@ -0,0 +1,209 @@
+﻿public class NullCoalescenceAssignment
+{
+    public void NullCoalescenceAssignment_NotNull(string s1)
+    {
+        s1 ??= "N/A";
+        s1.ToString(); // Compliant
+    }
+
+    public void NullCoalescenceAssignment_Null(string s1)
+    {
+        s1 ??= null;
+        s1.ToString(); // FIXME Non-compliant
+    }
+
+    public void InsideIf(string str)
+    {
+        if (str != null)
+        {
+            str ??= null;
+            str.ToString(); // Compliant, we know str is not null
+        }
+
+        if (str == null)
+        {
+            str ??= "foo";
+            str.ToString(); // Compliant, assigned foo
+        }
+    }
+}
+
+public interface IWithDefaultMembers
+{
+    decimal Count { get; set; }
+    decimal Price { get; set; }
+
+    void Reset(string s)
+    {
+        s.ToString(); // FIXME Non-compliant
+    }
+}
+
+public class LocalStaticFunctions
+{
+    public void Method(object arg)
+    {
+        string LocalFunction(object o)
+        {
+            return o.ToString(); // Compliant - FN: local functions are not supported by the CFG
```

Comment:
Is it "compliant" so we should never raise? Therefore it complies with the rule specification.
Or is it "FN" and we should fix it one day?
These two words are contradictory

Reply:
Marked as FN.